src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/DispatchOperation.cs

```diff
@@ -188,7 +188,7 @@ public string Name
             get { return name; }
         }
 
-        internal SynchronizedCollection<IParameterInspector> ParameterInspectors { get; }
+        public SynchronizedCollection<IParameterInspector> ParameterInspectors { get; set; }
 
         public DispatchRuntime Parent { get; }
```

Comment:
Does the setter provide any additional value? Or is it enough to be able to modify the collection?

Reply:
You are right, the setter is not needed. I updated the code to remove it.

Reply:
@Treit, I should have looked at this property more closely as it's an interesting one. The client corresponding equivalent of `DispatchOperation` is `ClientOperation`. At some point `ClientOperation` was modified to hide the `ParameterInspectors` property and added a new one called `ClientParameterInspectors` which was declared as `ICollection<IParameterInspector>`, ie:
```c#
        public ICollection<IParameterInspector> ClientParameterInspectors { get; }

        [EditorBrowsable(EditorBrowsableState.Never)]
        public new SynchronizedCollection<IParameterInspector> ParameterInspectors { get; }
```

I'm not suggesting a fully equivalent change here, but I do see there being value in changing the type of `DispatchOperation.ParameterInspectors` to be `ICollection<IParameterInspector>`. While that's technically a breaking change, moving to CoreWCF is technically one big breaking change anyway with the namespace change. I'm not seeing anything on `SynchronizedCollection` which isn't declared in one of the collection interfaces so I don't think this would cause any problems for anyone.
So I'm soft suggesting a change but asking for opinions. @Daniel-Svensson, what are your thoughts on my suggestion? The original comment was made against the wrong line to make a code suggestion so I'll make another comment with the suggestion so you can one-click apply.

Reply:
Sounds ok with ICollection as long as SynchronizedCollection does not imply some very specific behavior (I expect it provides some thread safety?) in which case it might make more sense to add a comment about that behavior instead. I se no really strong arguments for changing or keeping as is. 

pro: I like that it mirrors the client behavior, should the name also be changed to mimic *Client*ParameterInspectors? (or is it be better of changing the client to only have a single public `ParameterInspectors`). 

cons: The downside I see is that you will not be able to implicitly cast to IReadOnly.. or ICollection interfaces, but that is a somewhat minor issue.